internal/provider/waypoint/resource_waypoint_add_on_definition.go

```diff
@@ -0,0 +1,563 @@
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *AddOnDefinitionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan *AddOnDefinitionResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+
+	client := r.client
+	ns, err := getNamespaceByLocation(ctx, client, loc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error getting namespace by location",
+			err.Error(),
+		)
+		return
+	}
+
+	stringLabels := []string{}
+	if !plan.Labels.IsNull() && !plan.Labels.IsUnknown() {
+		diagnostics := plan.Labels.ElementsAs(ctx, &stringLabels, false)
+		if diagnostics.HasError() {
+			resp.Diagnostics.AddError(
+				"error converting labels",
+				"Failed to convert labels from types.List to string list",
+			)
+			return
+		}
+	}
+
+	readmeBytes, err := base64.StdEncoding.DecodeString(plan.ReadmeMarkdownTemplate.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error decoding base64 readme markdown template",
+			err.Error(),
+		)
+	}
+
+	modelBody := &waypointmodels.HashicorpCloudWaypointWaypointServiceCreateAddOnDefinitionBody{
+		Name:                   plan.Name.ValueString(),
+		Summary:                plan.Summary.ValueString(),
+		Description:            plan.Description.ValueString(),
+		ReadmeMarkdownTemplate: readmeBytes,
+		Labels:                 stringLabels,
+		TerraformNocodeModule: &waypointmodels.HashicorpCloudWaypointTerraformNocodeModule{
+			// verify these exist in the file
+			Source:  plan.TerraformNoCodeModule.Source.ValueString(),
+			Version: plan.TerraformNoCodeModule.Version.ValueString(),
+		},
+		TerraformCloudWorkspaceDetails: &waypointmodels.HashicorpCloudWaypointTerraformCloudWorkspaceDetails{
+			Name:      plan.TerraformCloudWorkspace.Name.ValueString(),
+			ProjectID: plan.TerraformCloudWorkspace.TerraformProjectID.ValueString(),
+		},
+	}
+
+	params := &waypoint_service.WaypointServiceCreateAddOnDefinitionParams{
+		NamespaceID: ns.ID,
+		Body:        modelBody,
+	}
+	def, err := r.client.Waypoint.WaypointServiceCreateAddOnDefinition(params, nil)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating add-on definition", err.Error())
+		return
+	}
+
+	var addOnDefinition *waypointmodels.HashicorpCloudWaypointAddOnDefinition
+	if def.Payload != nil {
+		addOnDefinition = def.Payload.AddOnDefinition
+	}
+	if addOnDefinition == nil {
+		resp.Diagnostics.AddError("unknown error creating add-on definition", "empty add-on definition found")
```

Comment:
```suggestion
		resp.Diagnostics.AddError("unknown error creating add-on definition", "empty add-on definition returned")
```

I'll change this in the other data source if that's what is says; having `found` here suggests to me it's something with the configuration, and not with what the server returned which is the intent.